### PR TITLE
Adjust folder structure to be component based

### PR DESCRIPTION
**Pull request by MrMnt**
Adjust folder structure to be component based instead of everything in /Repository, /Enums, /Controllers etc
Updated database migrations to reflect adjusted entities namespaces

### CODE REVIEW

**Comment by MrMnt** [comment]
Idea is that each component will have it's own Controller, Entities, Enums, Models and Repositories directory, for example
image

File: OmgvaPOS/src/UserManagement/Repository/UserRepository.cs [hunk 1]
```diff
@@ -97,7 +98,7 @@ public List<EmployeeSchedule> GetUserSchedules(long id)
                 throw new ApplicationException("Error retrieving user schedules.", ex);
             }
         }
-        public List<Order> GetUserOrders(long id)
+        public List<Order.Entities.OrderEntity> GetUserOrders(long id)
```
Comment (AugustinasJarockis) [inline OmgvaPOS/src/UserManagement/Repository/UserRepository.cs:101]: Why can't you just write OrderEntity? This is very verbose.

File: OmgvaPOS/src/UserManagement/Controller/UserController.cs [hunk 2]
```diff
@@ -236,7 +238,7 @@ public IActionResult GetUserSchedules(long userId)
 
         [HttpGet("{userId}/order")]
         [Authorize(Roles = "Admin, Owner, Employee")]
-        [ProducesResponseType<List<Order>>(StatusCodes.Status200OK)]
+        [ProducesResponseType<List<Order.Entities.OrderEntity>>(StatusCodes.Status200OK)]
```
Comment (AugustinasJarockis) [inline OmgvaPOS/src/UserManagement/Controller/UserController.cs:241]: Same here

File: OmgvaPOS/src/UserManagement/Controller/UserController.cs [hunk 3]
```diff
@@ -1,26 +1,28 @@
...
-    public class UserController(IAuthenticationRepository authenticationRepository, IUserRepository userRepository, ILogger<UserController> logger) : Controller
+    public class UserController(IAuthenticationRepository authenticationRepository, IUserRepository userRepository, ILogger<UserController> logger) : Microsoft.AspNetCore.Mvc.Controller
     {
         private readonly IAuthenticationRepository _authenticationRepository = authenticationRepository;
         private readonly IUserRepository _userRepository = userRepository;
         private readonly ILogger<UserController> _logger = logger;
 
         [HttpPost]
-        [ProducesResponseType<User>(StatusCodes.Status201Created)]
+        [ProducesResponseType<Entities.UserEntity>(StatusCodes.Status201Created)]
```
Comment (AugustinasJarockis) [inline OmgvaPOS/src/UserManagement/Controller/UserController.cs:25]: And here

File: OmgvaPOS/src/BusinessManagement/Controller/BusinessController.cs [hunk 4]
```diff
@@ -1,18 +1,18 @@
...
+using OmgvaPOS.UserManagement.Enums;
+using OmgvaPOS.Validators;
 
-namespace OMGVA_PoS.Business_layer.Controllers
+namespace OmgvaPOS.BusinessManagement.Controller
 {
     [Route("business")]
     [ApiController]
-    public class BusinessController(IBusinessRepository businessRepository, ILogger<BusinessController> logger) : Controller
+    public class BusinessController(IBusinessRepository businessRepository, ILogger<BusinessController> logger) : Microsoft.AspNetCore.Mvc.Controller
```
Comment (AugustinasJarockis) [inline OmgvaPOS/src/BusinessManagement/Controller/BusinessController.cs:15]: Doesn't just Controller work here?

**Review by AugustinasJarockis — COMMENTED** [review]
It is a bit strange that our models were renamed to entities and DTOs renamed to models, but whatever. Not approving yet because the approval will disspear after merge anyways.

File: OmgvaPOS/src/UserManagement/Repository/UserRepository.cs (hunk 1 above)
Comment (MrMnt) [reply OmgvaPOS/src/UserManagement/Repository/UserRepository.cs:101]: My bad, this was due to restructuring and happened automatically, tried to fix most of these, thanks for noticing

File: OmgvaPOS/src/UserManagement/Repository/UserRepository.cs [hunk 5]
```diff
@@ -98,7 +99,7 @@ public List<EmployeeScheduleEntity> GetUserSchedules(long id)
                 throw new ApplicationException("Error retrieving user schedules.", ex);
             }
         }
-        public List<Order.Entities.OrderEntity> GetUserOrders(long id)
+        public List<OrderManagement.Models.Order> GetUserOrders(long id)
```
Comment (AugustinasJarockis) [inline OmgvaPOS/src/UserManagement/Repository/UserRepository.cs:102]: The problem persists here.

**Review by AugustinasJarockis — DISMISSED** [review]
Overall, good.

File: OmgvaPOS/src/BusinessManagement/Controller/BusinessController.cs (hunk 4 above)
Comment (AugustinasJarockis) [reply OmgvaPOS/src/BusinessManagement/Controller/BusinessController.cs:15]: Apparently no.

**Review by AugustinasJarockis — APPROVED** [review]

**Review by GitaGliaudyte — APPROVED** [review]

**Pull request merged**